packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx

```diff
@@ -124,6 +216,47 @@ export const PaymentUpdateForm = ({
           </AlertBar>
         )}
 
+        {paypal_payment_error && (
+          <AlertBar className="alert alertError">
+            {errorAlertBarContent}
+          </AlertBar>
+        )}
+
+        {fixPaymentModalRevealed && (
+          <DialogMessage
+            data-testid="billing-info-modal"
+            onDismiss={hideFixPaymentModal}
+            className="billing-info-modal"
+          >
+            <img id="error-icon" src={errorIcon} alt="error icon" />
+            <Localized id="sub-route-payment-modal-heading">
+              <h2>Invalid billing information</h2>
+            </Localized>
+            <Localized id="sub-route-payment-modal-copy">
+              <p>
+                There seems to be an error with your Paypal account, we need you
+                to take the necessary steps to resolve this payment issue.
+              </p>
+            </Localized>
+
+            {paypalScriptLoaded && (
+              <Suspense fallback={<div>Loading...</div>}>
+                <div className="paypal-button">
+                  <PaypalButton
+                    currencyCode={currencyCode}
+                    customer={customer}
+                    idempotencyKey={submitNonce}
+                    priceId={priceId}
+                    refreshSubscriptions={refreshSubscriptions}
+                    setPaymentError={setPaymentError}
+                    ButtonBase={paypalButtonBase}
+                  />
```

Comment:
@bbangert Looking at the [Checkout ladder diagram](https://mozilla.github.io/ecosystem-platform/docs/fxa-engineering/fxa-sub-platform#paypal-checkout), is it okay to use the same endpoint (`POST /oauth/subscription/active/new-paypal`) here for `onApprove` in our `PaypalButton`?

In this case, we don't want to create a new subscription; we want to update an existing subscription... Is there different accounting for this in the back-end? Do we want to send the same emails as for a new subscription (i.e. [Welcome and First Invoice e-mails](https://miro.com/app/board/o9J_lWaWaZA=/))?

Reply:
@biancadanforth That's a great point. CC/ @bbangert Do you know if we have an answer for this currently?

Reply:
Yes, we do have a new endpoint for this condition that just updates the payment:
https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts#L356